File: exercise/24/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode fake(0);
        ListNode *point = &fake;
        fake->next = head;
        while (point->next && point->next->next) {
            ListNode *first = point->next;
            point->next = first->next;
            first->next = point->next->next;
            point->next->next = first;
            point = first->next;
        }
        return fake.next;
    }
};

File: exercise/24/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        fake = ListNode(0)
        tail = fake
        while tail.next and tail.next.next:
            temp = tail.next
            tail.next = temp.next
            temp.next = tail.next.next
            tail.next.next = temp
            tail = temp
        return fake.next
